kord-extensions/src/main/kotlin/com/kotlindiscord/kord/extensions/checks/utils/Scheduler.kt

```diff
@@ -0,0 +1,90 @@
+package com.kotlindiscord.kord.extensions.checks.utils
+
+import kotlinx.coroutines.*
+import mu.KotlinLogging
+import kotlin.coroutines.EmptyCoroutineContext
+
+private val logger = KotlinLogging.logger {  }
+
+/**
+ * Class in charge of providing scheduling functions.
+ */
+class Scheduler {
+    private val jobMap: MutableMap<Int, Job> = mutableMapOf()
+
+    private val scope = CoroutineScope(EmptyCoroutineContext)
```

Comment:
GlobalScope is just fine, as @BartArys has stated - but if you feel like it would be useful, keep the variable and make this an `open class` so that users can subclass it for their own needs.

I'm not sure how useful that'd be, though.